docs/source/configuration.mdx

```diff
@@ -124,32 +135,177 @@ server:
+#
+# Layers are features (either native or extensions) which can be enabled for
+# each subgraph.  Common layers are the "headers*" layers, and an example can be
+# found for below.  For more information, see later in the documentation.
+#
+
 subgraphs:
-  # Defines a subgraph named "accounts"
+  # Specifies the runtime configuration for talking to the "accounts" subgraph.
   accounts:
-    # The URL of the accounts subgraph's GraphQL endpoint
+    # The URL of the "accounts" subgraph's GraphQL endpoint
     routing_url: http://localhost:4001/graphql
-  # Defines a second subgraph named "products"
+
+    # Layers to be applied to the "accounts" subgraph.
+    layers:
+      - headers_propagate:
+          matching:
+            regex: ^upstream-header-.*
+      - headers_propagate:
+          named:
+            name: "old-name"
+            rename: "new-name"
+            default_value: "0"
+      - headers_propagate:
+          named:
+            name: "old-name"
+            rename: "new-name"
+            default_value: "0"
+      - headers_remove:
+          name: "x-legacy-account-id"
+      - headers_remove:
+          matching:
+            regex: ^x-deprecated.*
+      - headers_insert:
+          name: "router-subgraph-name"
+          value: "accounts"
+
+  # Specifies the runtime configuration for talking to the "products" subgraph.
   products:
+    # The URL of the "products" subgraph's GraphQL endpoint
     routing_url: http://localhost:4003/graphql
 
-# graph configuration, see later section
+    # Layers to be applied to the "products" subgraph.
+    layers:
+      - headers_insert:
+          name: "router-subgraph-name"
+          value: "products"
+      - headers_remove:
+          name: "x-legacy-tracking-id"
+
+
+# graph configuration, see documentation below
 # graph:
 
-# spaceport configuration, see later section
+# spaceport configuration, see documentation below
 # spaceport:
 
-# OpenTelemetry configuration, see next section
+# OpenTelemetry configuration, see documentation below
 # opentelemetry:
 ```
 
+## Configuring headers received by subgraphs
+
+Passing headers is supported by leveraging pre-built functionality in the Router known as "layers".  They are therefore defined in the pipeline as members of the named `subgraphs` member (e.g., `reviews`, `products`; often, certain headers only go to certain services) and further defined within the `layers` array.  The `layers` are an array since they're applied in an ordered fashion and later layers may override a previous layer's actions.
+
+There are a few specific layers which offer different header functionality:
+
+- **`headers_propagate`**
+
+  This allows _selectively_ adding headers which were client-originated to the requests made to _specific_ subgraphs.
+
+  They can be configured based on a `matching` pattern using [`regex`]:
+
+  ```yaml
+      - headers_propagate:
+          matching:
+            regex: .*
```

Comment:
Let's mention that host, content-length and content-type will not be propagate by regex:
https://github.com/apollographql/router/pull/505

Reply:
Are we case insensitive in our processing?

Reply:
it's compared with the http crate's `HeaderName`, which should handle that for us

Reply:
I'll link to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#hop-by-hop_headers

Reply:
I added